org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/editor/WRecordIDEditor.java

```diff
@@ -0,0 +1,367 @@
+			
+			if(tableIDGridField.getValue() != null)
+				tableIDEditor.setValue(tableIDGridField.getValue());
+						
+			dataBinder = new GridTabDataBinder(gridTab);
+			tableIDEditor.addValueChangeListener(dataBinder);
+			
+			tableIDGridField.addPropertyChangeListener(tableIDEditor);
+			tableIDGridField.addPropertyChangeListener(new PropertyChangeListener() {
+				@Override
+				public void propertyChange(PropertyChangeEvent evt) {
+					if(evt.getNewValue() != oldTableID) {
+						oldTableID = evt.getNewValue();
+						setValue(null);
+					}
+				}
+			});
+			oldTableID = tableIDGridField.getValue();
+		}
+		
+		recordTextBox = new Textbox();
+		recordTextBox.setParent(wrapperDiv);
+		recordTextBox.setWidth("100%");
+		recordTextBox.setReadonly(true);
+		
+		editButton = new ToolBarButton();
+		editButton.setStyle("right: 21px;");
+		if (ThemeManager.isUseFontIconForImage())
+			editButton.setIconSclass("z-icon-Edit");
+		else
+			editButton.setImage(ThemeManager.getThemeResource(IMAGES_CONTEXT_EDIT_RECORD_PNG));
+		editButton.setParent(wrapperDiv);
+		editButton.addEventListener(Events.ON_CLICK, this);
+
+		zoomButton = new ToolBarButton();
+		if (ThemeManager.isUseFontIconForImage())
+			zoomButton.setIconSclass("z-icon-Zoom");
+		else
+			zoomButton.setImage(ThemeManager.getThemeResource(IMAGES_CONTEXT_ZOOM_PNG));
+		zoomButton.setParent(wrapperDiv);
+		zoomButton.addEventListener(Events.ON_CLICK, this);
+		
+		if (gridField != null)
+        {
+			popupMenu = new WEditorPopupMenu(true, true, isShowPreference());
+			popupMenu.addMenuListener(this);
+			addChangeLogMenu(popupMenu);
+        }
+	}
+	
+	public void actionRefresh(Object value)
+    {
+		setValue(value);
+    }
+	
+	@Override
+	public void actionZoom()
+	{
+		Integer tableID = (Integer) tableIDGridField.getValue();
+		Integer recordID = (Integer) gridTab.getValue("Record_ID");
+		if(tableID == null || tableID <= 0 || recordID == null || recordID <= 0)
+			return;
+		if (!MRole.getDefault().isTableAccess(tableID, false))
+			throw new AdempiereException(Msg.getMsg(Env.getCtx(), "AccessTableNoView"));
+		MTable savedTable = new MTable(Env.getCtx(), tableID, null);
+		String[] keyColumns = savedTable.getKeyColumns();
+		String keyColumn = null;
+		if(keyColumns.length > 0) {
+			keyColumn = keyColumns[0];	//	guess
+		}
+	
+		String where = keyColumn + "=" + recordID;
+		PO po = new Query(Env.getCtx(), savedTable, where, null).first();
+		boolean isSOTrx = true;
+		if(po.columnExists("IsSOTrx")) {
+			isSOTrx = po.get_ValueAsBoolean("IsSOTrx");
+		}
+		int AD_Window_ID;
+		if(isSOTrx)
+			AD_Window_ID = savedTable.getAD_Window_ID();
+		else
+			AD_Window_ID = savedTable.getPO_Window_ID();
+
+		if (AD_Window_ID > 0) {
+			if(!Util.isEmpty(keyColumn)) {
+				MQuery zoomQuery = new MQuery();
+				zoomQuery.addRestriction(keyColumn, MQuery.EQUAL, (Integer)po.get_Value(keyColumn));
+				zoomQuery.setRecordCount(1);
+	
+				AEnv.zoom(AD_Window_ID, zoomQuery);
+			}
+		}
```

Comment:
don't really understand why not use use AEnv.zoom(AD_Table_ID, Record_ID) here ?